### PR TITLE
docs: New page for Shopify

### PR DESCRIPTION
**Pull request by kevee**
## Description

Adds a doc page about how to get quickly setup with Shopify. This will eventually be part of a sub section in [Adding website functionality](https://www.gatsbyjs.org/docs/adding-website-functionality/).

## Related Issues

Related to #10418

### CODE REVIEW

File: docs/docs/building-an-ecommerce-site-with-shopify.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+---
+
+In this tutorial, you will setup a new Gatsby website that fetches product data from [Shopify](https://www.shopify.com). The site displays a list of all products on a product listing page, and a page for every product in the store.
+
+If you are already comfortable with Gatsby and Shopify, you might want to check out the [Gatsby Shopify starter](https://www.gatsbyjs.org/starters/AlexanderProd/gatsby-shopify-starter/), which provides many of the same features as this example.
+
+## Setting up your Shopify account
+
+1. Create a new [Shopify account](https://www.shopify.com) and store if you don't have one.
+2. Create a Private App in your store by navigating to **Apps**, then "Manage private apps"
```
Comment (muescha) [inline docs/docs/building-an-ecommerce-site-with-shopify.md]:
```suggestion
2. Create a private app in your store by navigating to `Apps`, then `Manage private apps`.
```

File: docs/docs/building-an-ecommerce-site-with-shopify.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+3. Create a new private app and enable the [Shopify Storefront API](https://help.shopify.com/en/api/storefront-api). Make sure to also grant access to read product and customer tags.
+
+## Set up the Gatsby Shopify plugin
+
+1. If you do not already have one ready, [create a Gatsby site](https://www.gatsbyjs.org/docs/quick-start).
+
+2. Install the `gatsby-source-shopify` plugin and `shopify-buy` package.
```
Comment (muescha) [inline docs/docs/building-an-ecommerce-site-with-shopify.md]:
```suggestion
2. Install the [`gatsby-source-shopify`](/packages/gatsby-source-shopify/) plugin and [`shopify-buy`](https://github.com/Shopify/js-buy-sdk) package.
```

Comment (muescha) [reply docs/docs/building-an-ecommerce-site-with-shopify.md]: There is also an https://www.gatsbyjs.org/packages/gatsby-plugin-shopify-buy/

Is this better or similar?

File: docs/docs/building-an-ecommerce-site-with-shopify.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    options: {
+      // The domain name of your Shopify shop.
+      shopName: `[some-shop].myshopify.com`,
+
+      // The storefront access token
+      accessToken: `[token]`,
+    },
+  },
+]
+```
```
Comment (muescha) [inline docs/docs/building-an-ecommerce-site-with-shopify.md]: I would also add here a notice about private accestoken And link to the environment variable page?

File: docs/docs/building-an-ecommerce-site-with-shopify.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+            amount
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+To add a simple page listing all products, add a new file at `/src/pages/products.js`.
```
Comment (muescha) [inline docs/docs/building-an-ecommerce-site-with-shopify.md]:
```suggestion
For a simple page listing with all products add a new file at `/src/pages/products.js`.
```

File: docs/docs/building-an-ecommerce-site-with-shopify.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+    }
+  }
+`
+```
+
+## Generating a page for each product
+
+You can [programatically create pages](/tutorial/part-seven/) in Gatsby for every product in your Shopify store.
+
+Create a template for your product pages by adding a new file, `/src/templates/product.js`.
```
Comment (muescha) [inline docs/docs/building-an-ecommerce-site-with-shopify.md]:
```suggestion
Create a template for your product pages by adding a new file `/src/templates/product.js`.
```

**Comment by muescha** [comment]
The `shopify-buy` is not implemented or used in this guide?

**Comment by kevee** [comment]
Thanks for the feedback, @muescha - I feel like shopify-buy might need to stay out of the guide just to keep things simple. Especially as the starter already includes it, and shopify-buy is not really gatsby-specific.

**Pull request merged**